cmd/juju/application/register.go

```diff
@@ -52,11 +52,16 @@ func (r *RegisterMeteredCharm) RunPre(api MeteredDeployAPI, bakeryClient *httpba
 	if !metered {
 		return nil
 	}
+	return errors.Trace(r.RunPreBundle(api, bakeryClient, ctx, deployInfo))
+}
+
+func (r *RegisterMeteredCharm) RunPreBundle(api MeteredDeployAPI, bakeryClient *httpbakery.Client, ctx *cmd.Context, deployInfo DeploymentInfo) error {
 	info := deployInfo.CharmInfo
-	if r.Plan == "" && info.Metrics != nil && !info.Metrics.PlanRequired() {
+	if r.Plan == "" && info != nil && info.Metrics != nil && !info.Metrics.PlanRequired() {
```

Comment:
Hmm.. overloading logic here, `deployInfo.CharmInfo == nil` to mean "ignore the plan required checking logic we'd use otherwise" isn't very clear.

Reply:
I think only RunPre cares about this check.

Reply:
I'd also argue that only RunPre cares about ignoring `registerMetrics` errors for non-charmstore deployments.